File: examples/handler.py
import os
import json
from base64 import b64decode

import boto3

warm_secret = None
def kms_decrypt(secretName):
    client = boto3.client('kms')
    with open(os.path.join(os.environ["LAMBDA_TASK_ROOT"], "secret-baker-secrets.json")) as f:
        secrets = json.load(f)
    context = {"PARAMETER_ARN": secrets[secretName]["arn"]}
    resp = client.decrypt(CiphertextBlob=b64decode(secrets[secretName]["ciphertext"]), EncryptionContext=context)
    return resp['Plaintext'].decode('UTF-8')

SECRETS = [
    "MY_SECRET",
    "MY_OTHER_SECRET",
    "CUSTOM_SECRET",
]

def hello(event, context):
    output_builder = []
    for secret in SECRETS:
        value = kms_decrypt(secret)
        output_builder.append("%s : %s\n" % (secret, value))
    return "".join(output_builder)

File: examples/handler.js
"use strict";

const aws = require("aws-sdk");
const fs = require("fs");
const path = require("path");

async function decryptSecret(secretName) {
  let kms = new aws.KMS();

  const secretsFilePath = path.join(
    process.env.LAMBDA_TASK_ROOT,
    "secret-baker-secrets.json"
  );
  const file = fs.readFileSync(secretsFilePath);
  const secrets = JSON.parse(file);
  const params = {
    CiphertextBlob: Buffer.from(secrets[secretName]["ciphertext"], "base64"),
    EncryptionContext: { PARAMETER_ARN: secrets[secretName]["arn"] }
  };
  const response = await kms.decrypt(params).promise();
  return response.Plaintext.toString("ascii");
}

module.exports.hello = async (event, context) => {
  const secrets = [
    "MY_SECRET",
    "MY_OTHER_SECRET",
    "CUSTOM_SECRET"
  ];
  let output = "";
  try {
    for (const secret of secrets) {
        const value = await decryptSecret(secret);
        output = output + `Secret ${secret}: ${value.slice(0, 3)}...\n`;
    }
  } catch (error) {
    return `ERROR!: ${error}`;
  }
  return output;
};
